app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,52 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show edit update destroy]
+
+  UNITS = %w[teaspoon tablespoon cup ounce pound kilogram gram mililiter liter miligram
+             units].sort
+
+  # GET /foods or /foods.json
+  def index
+    @foods = Food.all.includes(:user)
+  end
+
+  # GET /foods/1 or /foods/1.json
+  def show; end
+
+  # GET /foods/new
+  def new
+    @food = Food.new
+  end
+
+  # GET /foods/1/edit
+  def edit; end
+
+  # POST /foods or /foods.json
+  def create
+    @food = Food.new(food_params)
+
+    if @food.save
+      redirect_to foods_path, notice: 'Food was successfully created.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
+    @food.destroy
+    redirect_to foods_path, notice: 'Food was successfully destroyed.'
+  end
```

Comment:
- [ ] I get the below error when I try to delete a food I added. Please kindly fix this.

![image](https://user-images.githubusercontent.com/43995830/175455617-db12a914-6cab-40c9-ae6c-5ad5bf61daf0.png)